Formula/mingw-w64.rb

```diff
@@ -0,0 +1,303 @@
+        end
+      end
+
+      ENV.prepend_path "PATH", bin.to_s
+
+      args = %W[
+        CC=#{target_arch}-gcc
+        CXX=#{target_arch}-g++
+        CPP=#{target_arch}-cpp
+        LD=#{target_arch}-gcc
+        --host=#{target_arch}
+        --prefix=#{prefix}/#{target_arch}
+        --with-sysroot=#{prefix}/#{target_arch}
+      ]
+      if target_arch.start_with?("x86_64")
+        args.push("--disable-lib32", "--enable-lib64")
+      elsif target_arch.start_with?("i686")
+        args.push("--enable-lib32", "--disable-lib64")
+      end
+      # elsif target_arch.start_with?("x86_64") and build.with?("multilib")
+      #  args.push("--enable-lib32", "--enable-lib64")
+
+      chdir "mingw-w64-crt" do
+        mkdir "build-#{target_arch}" do
+          system "../configure", *args
+          system "make", "-j1"
+          system "make", "install"
+        end
+      end
+
+      resource("gcc").stage do
+        args = %W[
+          CC=gcc-6
+          CXX=g++-6
+          CPP=cpp-6
+          LD=gcc-6
+          --build=#{arch}-apple-darwin#{osmajor}
+          --host=#{arch}-apple-darwin#{osmajor}
+          --target=#{target_arch}
+          --prefix=#{prefix}
+          --with-sysroot=#{prefix}
+          --disable-nls
+          --with-system-zlib
+          --enable-version-specific-runtime-libs
+          --enable-libstdcxx-time=yes
+          --enable-stage1-checking
+          --enable-checking=release
+          --enable-lto
+          --enable-libssp
+          --enable-libcc1
+          --enable-shared
+          --with-dwarf
+          --enable-fully-dynamic-string
+          --with-bugurl=https://github.com/Homebrew/homebrew-core/issues
+          --enable-threads=win32
+          --enable-languages=c,c++,objc,obj-c++
+          --with-gnu-as
+          --with-gnu-ld
+          --with-ld=#{bin}/#{target_arch}-ld
+          --with-as=#{bin}/#{target_arch}-as
+          --with-gmp=#{Formula["gmp"].opt_prefix}
+          --with-mpfr=#{Formula["mpfr"].opt_prefix}
+          --with-mpc=#{Formula["libmpc"].opt_prefix}
+          --with-cloog=#{Formula["cloog"].opt_prefix}
+          --with-isl=#{Formula["isl"].opt_prefix}
+          MAKEINFO=#{Formula["texinfo"].opt_bin}/makeinfo
+        ]
+        if target_arch.start_with?("i686")
+          args.insert(7, "--disable-multilib")
+          args.insert(15, "--disable-sjlj-exceptions")
+        elsif target_arch.start_with?("x86_64")
+          args.insert(7, "--enable-64bit")
+          args.insert(7, "--disable-multilib")
+        end
+        # elsif target_arch.start_with?("x86_64") and build.with?("multilib")
+        #  args.insert(7, "--enable-64bit")
+        #  args.insert(7, "--enable-targets=x86_64-w64-mingw32,i686-w64-mingw32")
+        #  args.insert(7, "--enable-multilib")
+        mkdir "build-#{target_arch}" do
+          system "../configure", *args
+          system "make", "all-target-libgcc"
+          system "make", "install-target-libgcc"
```

Comment:
I do not think all-target-libgcc needs to be done first. A simple "make" then "make install" should work here.